bam_controls.js

```diff
@@ -0,0 +1,262 @@
+                    GRCh37 Exonic Regions
+                </div>
+                <div id="default-bedfile-button-grch38" class="file-selection-button" title="1000G human exome targets file">
+                    GRCh38 Exonic Regions
+                </div>
+                <div class="iobio-file-picker" title="Add Bed format capture target definition file">
+                    <input type="file" id="file-selection" multiple>
+                    <label for="file-selection" class="file-selection-button">Custom Bed</label>
+                </div>
+            </div>
+            <button id="remove-bedfile-button">Remove</button>
+        </div>
+    </div>
+`;
+
+class BamControls extends HTMLElement {
+    constructor() {
+        super();
+        this.attachShadow({ mode: 'open' });
+        this.shadowRoot.appendChild(template.content.cloneNode(true));
+        this.initDOMElements();
+    }
+
+    initDOMElements() {
+        this.goButton = this.shadowRoot.querySelector('#bamview-controls-go');
+        this.searchButton = this.shadowRoot.querySelector('#gene-search-button');
+        this.chromosomeInput = this.shadowRoot.querySelector('#bamview-region-chromosome');
+        this.startInput = this.shadowRoot.querySelector('#bamview-region-start');
+        this.endInput = this.shadowRoot.querySelector('#bamview-region-end');
+        this.geneNameInput = this.shadowRoot.querySelector('#gene-name-input');
+        this.sourceSelect = this.shadowRoot.querySelector('#source-select');
+
+        this.goButton.addEventListener('click', () => this.handleGoClick());
+        this.searchButton.addEventListener('click', () => this.handleSearchClick());
+
+        this.defaultBedFileGRCh37 = this.shadowRoot.querySelector('#default-bedfile-button-grch37');
+        this.defaultBedFileGRCh38 = this.shadowRoot.querySelector('#default-bedfile-button-grch38');
+       
+        this.filePicker = this.shadowRoot.querySelector('#file-selection');
+        this.label = this.shadowRoot.querySelector('label');
+
+        this.removeBedFile = this.shadowRoot.querySelector('#remove-bedfile-button');
+    }
+
+    async connectedCallback() {
+        this.defaultBedFileGRCh37.addEventListener("click", () => this.handleBedfileClick('/grch37.bed', this.defaultBedFileGRCh37));
+        this.defaultBedFileGRCh38.addEventListener("click", () => this.handleBedfileClick('/grch38.bed', this.defaultBedFileGRCh38));
+        this.filePicker.addEventListener('change', (event) => this.handleBedfilePick(event));
+        this.removeBedFile.addEventListener("click", () => this.handleBedfileRemove('bedFileRemoved'));
+    }
+
+    handleGoClick() {
+        const chromosome = this.chromosomeInput.value.trim();
+        const start = this.startInput.value.trim();
+        const end = this.endInput.value.trim();
+
+        const event = new CustomEvent('go-click', {
```

Comment:
I think the default input will be string type. 
And if we use the Number() to convert the empty string, the result will be `0` instead of `undefined`. The safety check you mentioned in the comment below in `BamView` is for checking the scenario if they're empty string. If they're, then it will call the function `zoomToChromosome`, if not and they're both integers, it will call the function `brushToRegion`

Reply:
start and end should never be String except when they are entered in the inputs. You should convert them here, as soon as you pull them out of the inputs. That way they will always be Number later. If you need checks do them here.